glue/core/roi.py

```diff
@@ -544,6 +543,90 @@ def __setgluestate__(cls, rec, context):
         return cls(xc=rec['xc'], yc=rec['yc'], radius=rec['radius'])
 
 
+class CircularAnnulusROI(Roi):
+    """
+    A 2D circular annulus region of interest.
+
+    Parameters
+    ----------
+    xc : float, optional
+        `x` coordinate of center.
+    yc : float, optional
+        `y` coordinate of center.
+    inner_radius : float, optional
+        Inner radius of the circular annulus.
+    outer_radius : float, optional
+        Outer radius of the circular annulus.
+    """
+    def __init__(self, xc=None, yc=None, inner_radius=None, outer_radius=None):
+        super().__init__()
+        self.xc = xc
+        self.yc = yc
+        self.inner_radius = inner_radius
+        self.outer_radius = outer_radius
+
+    def contains(self, x, y):
+        if not self.defined():
+            raise UndefinedROI
+
+        if not isinstance(x, np.ndarray):
+            x = np.asarray(x)
+        if not isinstance(y, np.ndarray):
+            y = np.asarray(y)
+
+        dx = x - self.xc
+        dy = y - self.yc
+        r = np.sqrt((dx * dx) + (dy * dy))
+        # Python likes to do inclusive for min limit and exclusive for max limit, so why not.
+        return (r >= self.inner_radius) & (r < self.outer_radius)
+
+    def reset(self):
+        """Reset the circular annulus region."""
+        self.xc = None
+        self.yc = None
+        self.inner_radius = None
+        self.outer_radius = None
+
+    def defined(self):
+        number = (float, int)
+        if (isinstance(self.xc, number) and isinstance(self.yc, number) and
+                isinstance(self.inner_radius, number) and isinstance(self.outer_radius, number) and
+                (self.inner_radius > 0) and (self.outer_radius > self.inner_radius)):
+            status = True
+        else:
+            status = False
+        return status
+
+    # FIXME: We need 2 polygons, with the inner one inverted, just like subset state.
+    def to_polygon(self):
+        if not self.defined():
+            return [], []
+        raise NotImplementedError("Cannot transform annulus to simple polygon")
```

Comment:
Maybe better than nothing would be to transform it to a polygon like this:

<img width="428" alt="Screenshot 2023-05-12 at 10 05 18" src="https://github.com/glue-viz/glue/assets/314716/c2232a87-01e1-4aa0-8fa4-7b046a073f09">

I've exaggerated the separation of the lines at the top, these could be actually over each other if you see what I mean. Does this make sense?